### PR TITLE
Fix poll update handler calling method was that was not available

### LINKED ISSUE #10241: An error occurs in the notification of poll
### Actual behaviour

Since #10209 , creating a poll will result in errors on the server that created it and on the server that voted.

The side that made the poll:

- Job: ActivityPub::DeliveryWorker
- Error Class: ArgumentError
- Error Message: Delivery failed for : ArgumentError

The side who voted:

- Job: ActivityPub::ProcessingWorker
- Error Class: NoMethodError
- Error Message: undefined method `invalid_origin?' for #<ActivityPub::Activity::Update:0x000056551b494af0>
- Stack trace is [here](https://gist.github.com/mayaeh/2569bceac1d5579faf2c208ece2508b2)

### Steps to reproduce the problem

Make a poll, vote on another server and wait until the deadline.

### Specifications

equivalent Mastodon master 85537b00695b1091e2a147597ab7c3557b150b11

### PR DESCRIPTION
**Pull request by Gargron**
Fix #10241

### CODE REVIEW

**Review by ClearlyClaire — APPROVED** [review]

**Comment by Gargron** [comment]
@ThibG I have not found the source for the DeliveryWorker problem yet :confused:

**Comment by ThibG** [comment]
@Gargron me neither

File: app/lib/activitypub/activity/update.rb [hunk 1]
```diff
@@ -18,10 +21,9 @@ def update_account
...
   def update_poll
     return reject_payload! if invalid_origin?(@object['id'])
+
     status = Status.find_by(uri: object_uri, account_id: @account.id)
-    return if status.nil? || status.poll_id.nil?
-    poll = Poll.find(status.poll_id)
-    return if poll.nil?
+    return if status.nil? || status.poll.nil?
 
     ActivityPub::ProcessPollService.new.call(poll, @object)
```
Comment (ThibG) [inline app/lib/activitypub/activity/update.rb]: `poll` isn't defined anymore now, is it?

**Review by ClearlyClaire — CHANGES_REQUESTED** [review]
I think you overlooked something (inline comment).
Otherwise LGTM.

File: app/services/vote_service.rb [hunk 2]
```diff
@@ -20,21 +20,35 @@ def call(account, poll, choices)
     end
 
     if @poll.account.local?
-      ActivityPub::DistributePollUpdateWorker.perform_in(3.minutes, @poll.status.id) unless @poll.hide_totals
+      notify_about_new_votes!
```
Comment (ThibG) [inline app/services/vote_service.rb]: `distribute_poll!`?

**Review by ClearlyClaire — APPROVED** [review]

**Pull request merged**